### PR TITLE
Feature task2 program

### PR DESCRIPTION
**Pull request by MarkoAheienkov**
Program for manipulate data

### CODE REVIEW

File: Classes/Program/index.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+option=sort selector=s order=-1 - show sorted array in decrease order
+option=sort selector=s page=2 - show next 100 sorted records
+        `);
+  }
+  /**
+   * Log error
+   * @param {String} errorText - error text
+   */
+  logError(errorText) {
+    logger.color('red').log(errorText);
```
Comment (ostaplisovyj) [inline Classes/Program/index.js]: please extract logger to separate class and pass it to constructor as a dependency

File: Classes/FileSystem/FileSystemWriter/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * Create File System Writer
+ *
+ */
+class FileSystemWriter {
+  /**
+   * Method for write data to file
+   * @param {String} path
+   * @param {[*]} data
+   */
+  static async write(path, data) {
```
Comment (ostaplisovyj) [inline Classes/FileSystem/FileSystemWriter/index.js:11]: add `throw new Error()` to simulate abstract method

File: Classes/Program/index.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  async getData() {
+    return await FileSystemReader.read(this.pathToFile);
+  }
+  /**
+   * Log options
+   */
+  logOptions() {
+    logger.color('cyan').log('Options:');
+    logger.color('cyan').log(`
+option=help - show all operations
```
Comment (ostaplisovyj) [inline Classes/Program/index.js]: indent the following lines properly

Comment (ostaplisovyj) [inline Classes/Program/index.js]: indent the following lines properly

File: Classes/Program/index.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+   * Event listener for readLine
+   */
+  onLine() {
+    const option = this.getOptionFromInput();
+    this.callOption(option);
+  }
+  /**
+   * @return {String}
+   */
+  getOptionFromInput() {
```
Comment (ostaplisovyj) [inline Classes/Program/index.js]: if this method is private, prepend it with `_`

File: Classes/Program/index.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    return option;
+  }
+
+  /**
+   * @param {String} option - option
+   * @return {void}
+   */
+  callOption(option) {
+    if (!this.options[option]) {
+      this.logError(`No such option like: '${option}'`);
```
Comment (ostaplisovyj) [inline Classes/Program/index.js]: no need to use the separate method for logging error, see comment about logger class

File: Classes/Program/index.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+   * Main function
+   */
+  async main() {
+    this.data = await this.getData();
+    this.onLine();
+  }
+  /**
+   * Event listener for readLine
+   */
+  onLine() {
```
Comment (ostaplisovyj) [inline Classes/Program/index.js]: this is not exactly an event listener, rename to `processData()` or similar

File: Classes/Program/index.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+      this.show(this.data, limit, page);
+    }
+  }
+  /**
+   * Show all records
+   * @param {[]} data - data to show
+   */
+  showAll(data) {
+    for (let i = 0; i < data.length; i++) {
+      console.log(data[i]);
```
Comment (ostaplisovyj) [inline Classes/Program/index.js]: perhaps logger?

File: Classes/Program/index.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+      this.logError(`Page must be a number`);
+      return;
+    }
+    const from = limit*(page-1);
+    const to = limit*page;
+    for (let i = from; i < to; i++) {
+      if (!data[i]) {
+        break;
+      }
+      console.log(data[i]);
```
Comment (ostaplisovyj) [inline Classes/Program/index.js]: logger?

File: Scripts/writeDataFromApiToFiles/index.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+const PATH_TO_FILE = path.resolve(__dirname, '..', '..', 'Database', 'db.json');
+
+const getData = async (url) => {
+  const res = await fetch(url);
+  const data = await res.json();
+  return data;
+};
+
+const getDataAndWriteIt = async () => {
```
Comment (ostaplisovyj) [inline Scripts/writeDataFromApiToFiles/index.js]: let's split into two methods, one for getting the data and one for writing. Please think of more conventional names

**Review by ostaplisovyj — CHANGES_REQUESTED** [review]

File: Classes/Program/index.js (hunk 7 above)
Comment (MarkoAheienkov) [reply Classes/Program/index.js]: > perhaps logger?

You mean:
Make in Logger Class method to log array
Or
Just make not console.log but logger.log ?

(logger - instance of class Logger)

**Review by ostaplisovyj — APPROVED** [review]

**Pull request merged**